commit 7ab5c593f77b229210a88d436270707f74b22b78
Author: Michael Wallner <mike@php.net>
Date:   Thu Apr 3 09:07:35 2014 +0200

    Fix bug #66182 	exit in stream filter produces segfault
    
    Unfortunately, a segv caused by exit cannot be tested reliably.

--- php5.orig/ext/standard/user_filters.c
+++ php5/ext/standard/user_filters.c
@@ -180,6 +180,11 @@ php_stream_filter_status_t userfilter_fi
 	zval zpropname;
 	int call_result;
 
+	/* the userfilter object probably doesn't exist anymore */
+	if (CG(unclean_shutdown)) {
+		return ret;
+	}
+
 	if (FAILURE == zend_hash_find(Z_OBJPROP_P(obj), "stream", sizeof("stream"), (void**)&zstream)) {
 		/* Give the userfilter class a hook back to the stream */
 		ALLOC_INIT_ZVAL(zstream);
